phabricatoremails/mail.py

```diff
@@ -111,14 +113,22 @@ def send(self, emails: List[OutgoingEmail]):
                 f'[{email.to}] Sending "{email.template_path}" for "{email.subject}"'
             )
 
-            mime_message = email.to_mime_message(
-                self._from_address, include_target_in_subject=self._send_to is not None
-            )
-            self._server.sendmail(
-                self._from_address,
-                self._send_to if self._send_to else email.to,
-                mime_message.as_string(),
-            )
+            try:
+                mime_message = email.to_mime_message(
+                    self._from_address,
+                    include_target_in_subject=self._send_to is not None,
+                )
+                self._server.sendmail(
+                    self._from_address,
+                    self._send_to if self._send_to else email.to,
+                    mime_message.as_string(),
+                )
+            except Exception as e:
```

Comment:
Can we be more specific about which exceptions are being caught here?

Reply:
In theory, yes, but then if there's any exceptions that we miss, then that exception will bubble to the top, killing our application, and our queue will get "stuck" again.
Therefore, I think having a catch-all is a better trade-off here?

Note: I can make this more specific instead, then add a generic catch-all for the entire "batch" of events. But, if an unexpected exception is encountered here, then rather than skipping the one email/event, we will skip the rest of the "batch", causing additional lost emails.

Reply:
Generally, it's better to be specific about which exceptions we are catching (an example would be the common exceptions that you're trying to solve for in this PR). So what you've suggested would be good. If there are more cases that become known, we could add them to this later, or find other solutions.

When you say "additional lost emails" do you mean permanently?

Reply:
> When you say "additional lost emails" do you mean permanently?

Yes, those skipped emails will be lost permanently.

Reply:
If the loss of emails is critical, what we could do is catch the known exceptions in one block, and then the unknown ones after (and perhaps have a more specific message to differentiate those exceptions so we can deal with them later).